csm/conf/post_install.py

```diff
@@ -0,0 +1,177 @@
+        1. It first gets all the unsupported features lists of the components,
+        which CSM interacts with. Add all these features as CSM unsupported
+        features. The list of components, CSM interacts with, is
+        stored in csm.conf file. So if there is change in name of any
+        component, csm.conf file must be updated accordingly.
+        2. Installation/envioronment type and its mapping with CSM unsupported
+        features are maintained in unsupported_feature_schema. Based on the
+        installation/environment type received as argument, CSM unsupported
+        features can be stored.
+        """
+
+        def get_component_list_from_features_endpoints():
+            Log.info("Get Component List.")
+            feature_endpoints = Json(
+                const.FEATURE_ENDPOINT_MAPPING_SCHEMA).load()
+            component_list = [feature for v in feature_endpoints.values() for
+                              feature in v.get(const.DEPENDENT_ON)]
+            return list(set(component_list))
+        try:
+            Log.info("Set unsupported feature list to ES.")
+            # TODO: Add Proper Key as per Config Store
+            self._setup_info = Conf.get(const.CONSUMER_INDEX, const.GET_SETUP_INFO)
+            unsupported_feature_instance = unsupported_features.UnsupportedFeaturesDB()
+            components_list = get_component_list_from_features_endpoints()
+            unsupported_features_list = []
+            for component in components_list:
+                Log.info(f"Fetch Unsupported Features for {component}.")
+                unsupported = await unsupported_feature_instance.get_unsupported_features(
+                    component_name=component)
+                for feature in unsupported:
+                    unsupported_features_list.append(
+                        feature.get(const.FEATURE_NAME))
+            csm_unsupported_feature = Json(
+                const.UNSUPPORTED_FEATURE_SCHEMA).load()
+            for setup in csm_unsupported_feature[const.SETUP_TYPES]:
+                if setup[const.NAME] == self._setup_info[const.STORAGE_TYPE]:
+                    unsupported_features_list.extend(
+                        setup[const.UNSUPPORTED_FEATURES])
+            unsupported_features_list = list(set(unsupported_features_list))
+            unique_unsupported_features_list = list(
+                filter(None, unsupported_features_list))
+            if unique_unsupported_features_list:
+                Log.info("Store Unsupported Features.")
+                await unsupported_feature_instance.store_unsupported_features(
+                    component_name=str(const.CSM_COMPONENT_NAME),
+                    features=unique_unsupported_features_list)
+            else:
+                Log.info("Unsupported features list is empty.")
+        except Exception as e_:
+            Log.error(f"Error in storing unsupported features: {e_}")
```

Comment:
In case of error, we should throw exception so that caller can handle and print to user.

Reply:
sure will check and make necessary changes.

Reply:
Hi @ujjwalpl  Checked for the above issue with @soniyamoholkar. This is purposefully done since even if cortx-py-utils fails csm-agent should run irrespective of unsupported feature. 
Hence if an error is to arise we are just logging it.

Reply:
Done